src/main/scala/cognite/spark/v1/SequenceRowsRelation.scala

```diff
@@ -187,10 +199,15 @@ class SequenceRowsRelation(config: RelationConfig, sequenceId: CogniteId)(val sq
     } else {
       val (columns, fromRowFn) = fromRow(rows.head.schema)
       val projectedRows = rows.map(fromRowFn)
-
-      client.sequenceRows
-        .insertById(sequenceInfo.id, columns, projectedRows)
-        .flatTap(_ => incMetrics(itemsCreated, rows.length))
+      import cats.instances.list._
+      projectedRows
+        .groupBy(_.externalId)
+        .toList
+        .traverse(
+          pair =>
+            client.sequenceRows
+              .insertByExternalId(pair._1.toString, columns, pair._2)
+              .flatTap(_ => incMetrics(itemsCreated, rows.length))) *> IO.unit
```

Comment:
Nitpick

```suggestion
        .traverse {
          case (externalId, rows) =>
            client.sequenceRows
              .insertByExternalId(externalId.toString, columns, rows)
              .flatTap(_ => incMetrics(itemsCreated, rows.length)) } *> IO.unit
```

Reply:
Awesome, that's the one I was looking for!
Weirdly, `(externalId, rows)` typechecks, but then everything is in `externalId` and the `rows` is just junk I think